channels/chan_pjsip.c

```diff
@@ -2422,16 +2422,28 @@ static int call(void *data)
 static int chan_pjsip_call(struct ast_channel *ast, const char *dest, int timeout)
 {
 	struct ast_sip_channel_pvt *channel = ast_channel_tech_pvt(ast);
+	struct ast_sip_session *session;
+
 	SCOPE_ENTER(1, "%s Topology: %s\n", ast_sip_session_get_name(channel->session),
 		ast_str_tmp(256, ast_stream_topology_to_str(channel->session->pending_media_state->topology, &STR_TMP)));
 
 	ao2_ref(channel, +1);
-	if (ast_sip_push_task(channel->session->serializer, call, channel)) {
+
+	session = ao2_bump(channel->session);
+	ast_channel_unlock(ast);
+
+	/* The creation of the INVITE needs to be pushed synchronously to prevent a race condition
+	   with bridging on attended transfers that can result in a loss of set Caller ID. */
+	if (ast_sip_push_task_wait_serializer(session->serializer, call, channel)) {
```

Comment:
This assumes channel will remain valid for the lifetime of this, instead "session" should be passed and the "call" function be updated to use session .